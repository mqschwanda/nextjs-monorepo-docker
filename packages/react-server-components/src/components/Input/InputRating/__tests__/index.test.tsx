import { render } from '@testing-library/react';
import { InputRating } from '@';

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<InputRating />', () => {
      it('renders', () => {
        const testId = InputRating.name;

        const { getByTestId, asFragment } = render((
          <InputRating
            className='test'
            max={5}
            name='test-rating'
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });

      it('renders halves', () => {
        const testId = InputRating.name;

        const { getByTestId, asFragment } = render((
          <InputRating
            className='test'
            half
            max={5}
            name='test-rating'
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
