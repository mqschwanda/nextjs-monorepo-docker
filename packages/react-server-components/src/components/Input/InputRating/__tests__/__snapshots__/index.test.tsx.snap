// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <InputRating /> renders 1`] = `
<DocumentFragment>
  <div
    class="rating test"
    data-testid="InputRating"
  >
    <input
      class="mask rating-hidden mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-star"
      name="test-rating"
      type="radio"
    />
  </div>
</DocumentFragment>
`;

exports[`@mqs/react-server-components components <InputRating /> renders halves 1`] = `
<DocumentFragment>
  <div
    class="rating rating-half test"
    data-testid="InputRating"
  >
    <input
      class="mask rating-hidden mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-1 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-2 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-1 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-2 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-1 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-2 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-1 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-2 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-1 mask-star"
      name="test-rating"
      type="radio"
    />
    <input
      class="mask mask-half-2 mask-star"
      name="test-rating"
      type="radio"
    />
  </div>
</DocumentFragment>
`;
